code/controlconfig/controlsconfig.cpp

```diff
@@ -2689,6 +2710,28 @@ void control_get_axes_readings(int *axis_v, float frame_time)
 		if (!item.second.empty()) {
 			scale_invert(item.second, action, item.type, frame_time, axe, axis_v);
 		}
+
+		//Call Lua hooks
+		if (control_run_lua(static_cast<IoActionId>(action + JOY_AXIS_BEGIN), axis_v[action])) {
+			Assert(item.type == CC_TYPE_AXIS_ABS || item.type == CC_TYPE_AXIS_REL);
+
+			switch (item.type) {
+			case CC_TYPE_AXIS_ABS:
+				axis_v[action] = axes_last_value[action];
+				break;
+			case CC_TYPE_AXIS_REL:
+				axis_v[action] = 0;
+				break;
+			case CC_TYPE_TRIGGER:
+			case CC_TYPE_CONTINUOUS:
```

Comment:
I'm actually surprised none of the compilers are complaining about the lack of a default here.

Reply:
> I'm actually surprised none of the compilers are complaining about the lack of a default here.

Weird behaviour cause it's a switch on an enum, I think. It does however require me to include Trigger and Continuous, as otherwise the compiler will complain about incomplete switches on enums

Reply:
That's probably why the compiler isn't complaining about the missing `default` then, since you defined a case for every member of the enum.  Likewise, if you define a `default` case the compiler should by happy about the cases.

Reply:
I can add a default statement if you want, but I'd keep the CC_TYPE_AXIS_BTN_POS and CC_TYPE_AXIS_BTN_NEG cases, since they need to be thought about here, depending on implementation.

Reply:
Yes, both AXIS_BTN cases stay in.

Reply:
I thought about using that, decided against it, since the Assertion actually catches all cases currently leading to where the UNREACHABLE would be

Reply:
Ok, so its just a matter of style to use an assert before the switch or `UNREACHABLE` within the default in this case, which I'll leave up to you if you want to change or not. :)

Reply:
I'd like to keep it as an Assert, since a bad change further up the code could call this function with a different type, making it more of a safety check than a "this code is actually unreachable"-case. Also, it forces to acknowledge the Assert when implementing button axes and not just allows for a stray break in the switch case to solve that.